bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/items/events/ItemEventFactory.java

```diff
@@ -396,6 +351,7 @@ private static void assertValidArgument(Item item, String argumentName) {
     private static class ItemEventPayloadBean {
         private String type;
         private String value;
+        private Map<String, String> stateMap;
```

Comment:
agreed. This was the last resort to avoid having to deal with special state handling in the UI. I will review this again and come up with a better solution. The comment for `QuantityType#toFullString` is related. Should be fixed by a better solution, too.